src/TelemetryChannels/ServerTelemetryChannel/Shared/Implementation/TelemetryBuffer.cs

```diff
@@ -70,6 +72,29 @@ public int Capacity
             }
         }
 
+        /// <summary>
+        /// Gets or sets the maximum number of telemetry items that can be in the backlog to send. Items will be dropped
+        /// once this limit is hit.
+        /// </summary>
+        /// <exception cref="ArgumentOutOfRangeException">The value is zero or less.</exception>
+        public int MaximumBacklogSize
+        {
+            get
+            {
+                return this.maximumBacklogSize;
+            }
+
+            set
+            {
+                if (value < 1001)
```

Comment:
1001 is minimum customer can set. 1 million is the default. (for both inmem and server channel)